include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -419,30 +529,60 @@ __pattern_search(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _R1
 // search_n
 //------------------------------------------------------------------------
 
+template <typename _Tp>
+struct __pattern_search_n_fn
+{
+    const _Tp& __value;
```

Comment:
```suggestion
    _Tp __value;
```

Reply:
@MikeDvorskiy what do you think about this change?

Reply:
I cannot answer here by one line.
I need to see whole call/usage context.